control-plane/connect-inject/controllers/serviceaccount/serviceaccount_controller.go

```diff
@@ -155,9 +155,6 @@ func getWorkloadIdentityID(name, namespace, partition string) *pbresource.ID {
 		Tenancy: &pbresource.Tenancy{
 			Partition: partition,
 			Namespace: namespace,
-			// Because we are explicitly defining NS/partition, this will not default and must be explicit.
```

Comment:
FYI: this was probably a bug. Ideally tenancy defaulting will replace any empty fields regardless of others being defined.